tools/dataset_object_processor.py

```diff
@@ -51,7 +52,62 @@ def print_in_color(print_string="", color=PrintColors.WHITE) -> None:
     print(color + print_string + PrintColors.ENDC)
 
 
+def process_imported_asset(asset_path: str = "") -> None:
+    """
+    Use the trade.AbstractImporter class to query data size of mesh and image
+    of asset
+    """
+    # Open AbstractImporter
+    manager = trade.ImporterManager()
+    importer = manager.load_and_instantiate("AnySceneImporter")
+    importer.open_file(asset_path)
+    print_in_color(f"Plugin dir: {manager.plugin_directory}", PrintColors.PURPLE)
+    print_in_color("-" * 72, PrintColors.PURPLE)
+
+    # Get mesh data
+    print_in_color("\nMesh Data", PrintColors.GREEN)
+    print_in_color("-" * 72, PrintColors.GREEN)
+    print_in_color(f"Scene importer class: {type(importer)}", PrintColors.GREEN)
```

Comment:
Just FYI: I'm not exposing the particular importer plugin subclasses (like `GltfImporter`, `AssimpImporter`...) to Python (and I never plan to), so `type(importer)` would show the same (`magnum.trade.AbstractImporter`) for all plugins. The way to query which plugin class/module is actually used would be through [plugin()](https://doc.magnum.graphics/corrade/classCorrade_1_1PluginManager_1_1AbstractPlugin.html#a15b2a4e5e20275e08d4e5ad34b387e8b) (which isn't exposed to Python at the moment). In case of the `Any*Importer` it's further complicated because it delegates to a concrete plugin implementation internally. The [ImporterFlag::Verbose](https://doc.magnum.graphics/magnum/namespaceMagnum_1_1Trade.html#a137d832c7ce6e6b654022321941d70f3ad4a9fa383ab700c5bdd6f31cf7df0faf) (which... isn't exposed to Python at the moment either) makes it print the chosen plugin name to the output. TL;DR: there isn't much point in showing this particular piece of information.

To get a better idea of how the plugins get picked and behave I think it might be helpful for you to get the prebuilt tools binaries I linked on Slack and play with `magnum-sceneconverter --info <some-file.gltf>`. The `-I` option allows you to choose a particular importer plugin (`AnySceneImporter` is the default, which delegates to `GltfImporter` for glTF files, or you can specify `GltfImporter` directly, you can also pick `AssimpImporter` because Assimp sort-of knows glTF as well), `--verbose` will print additional information like what plugins got used to open a particular image etc. The `--importer-info` will then print info about a particular plugin specified with `-I`. For a full picture, try with `*.dae`, `*.ply` and `*.stl` files as well if you find them.

If you don't want to spend time reinventing things, you might possibly also want to steal from / get inspired from how the `magnum-sceneconverter --info` output looks like, what it contains and how is it formatted. Not everything is exposed to Python at the moment, but the essential stuff should be -- and I'm always happy to extend the bindings where needed.

*(Apologies if this is overwhelming, I'm just trying to share the info. If you feel it's too much and you don't need that many details, let me know ;))*

Reply:
No worries that was all good information. Yeah once I saw that the output is always `magnum.trade.AbstractImporter` I realized more how the Python side works. I assumed it was doing some delegating behind the scenes to determine the correct C++ importer class to use.

Reply:
All that delegating magic is inside the `Any*` plugins themselves -- if you use them instead of a concrete [plugin name or alias](https://doc.magnum.graphics/magnum/file-formats.html), they will attempt to detect based on file extension and delegate to *some* implementation for given format: https://github.com/mosra/magnum/blob/master/src/MagnumPlugins/AnySceneImporter/AnySceneImporter.cpp#L85-L144

In case of images, there's also detection based on file signatures, similarly to what the Unix `file` command does: https://github.com/mosra/magnum/blob/17638a2ad50094ece8eeb7ad7f9b06033d05d843/src/MagnumPlugins/AnyImageImporter/AnyImageImporter.cpp#L202-L270

Reply:
Yes, although there's no practical difference compared to using it through the AnySceneImporter.

Reply:
Is there any case where there WOULD be a practical difference? I also like my code, style-wise, to be very readable. Sometimes, if the filetype is very well known, I like to be explicit of what kind of variable/class an object is

Reply:
Okay, so the differences are:

- (the obvious one) if you have a *very minimal* build/installation that has only `GltfImporter` but no `AnySceneImporter` (or equivalently for other file types), you have to use `GltfImporter` directly
- the `AnySceneImporter` has some overhead compared to using `GltfImporter` directly because of the file format detection and delegation to another plugin (but nothing that would be in any way significant, so it's not a useful optimization strategy)
- if you would want to read the plugin-specific configuration for whatever reason, `GltfImporter` gives you back [all its options](https://doc.magnum.graphics/magnum/classMagnum_1_1Trade_1_1GltfImporter.html#Trade-GltfImporter-configuration), but `AnySceneImporter` will give you back an empty configuration, because on its own it doesn't have any option. Unlikely to affect practical use in code, but matters when you'd want to check the config options in [`magnum-sceneconverter --info-importer`](https://doc.magnum.graphics/magnum/magnum-sceneconverter.html#magnum-sceneconverter-example), for example.
- using `GltfImporter` instead of `AnySceneImporter` makes it clearer what you're trying to do, yes, you're right about that .. but OTOH it can be seen as hardcoding a specific file format, so when someone would want to use the same code for loading COLLADA or OBJ/PLY/STL... files in the future, they would have to change the code first